samples: lsm6dsl: Add a harness_config

erwango opened the pull request:
In order to enable sample to be run and evaluated in sanitycheck,
add a harness_config to validate sample output.
Tested on disco_l475_iot1

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

avisconti commented on samples/sensor/lsm6dsl/sample.yaml:
```diff
@@ -5,3 +5,9 @@ tests:
     tags: sensors
     depends_on: lsm6dsl
+    harness_config:
+      type: multi_line
```
Why not using the 'ordered' rule?
ordered: true

Maybe it does not add a big added value, but I would use it.
What do you think?

erwango replied:
Looking into it. So far, I don't manage to get test passing. Seems like regex not good enough.

avisconti reviewed (changes requested):
LGTM, but I was wondering if 'ordered' is worth to be used or not.
Anyway, I learnt something new which I might use to improve my samples

erwango commented:
Setting as DNM since I can't getting it passed.

avisconti commented:
> Setting as DNM since I can't getting it passed.

Do you mean that test is not passing if you use 'ordered'? Maybe we can just skip it for the moment. What do you think?

erwango committed:
samples: lsm6dsl: Add a harness_config

In order to enable sample to be run and evaluated in sanitycheck,
add a harness_config to validate sample output:
- Add "regex" to match on sample output
- Add "timeout" to save some time when debugging
- Add "ordered" instruction. Since sample is running a while loop
test verdict can potentially be computed on previous run output,
issuing a wrong status.

Last, since sanitycheck regex does not play well them, rework
sample to output without parenthesis.

Tested on disco_l475_iot1

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@avisconti, thanks!
I took me few hours to fix this but it was useful:
Sanitycheck regex does not play well with parenthesis, so I had to rework the harness.
Besides, since sample is running a while loop, test verdict could be done on console output acquired be fore flash completion, providing incoherent verdict.
Using "ordered: true" prevents this and is mandatory for this kind of tests.
I'm not sure how/why I initially saw the test PASSED despite the parenthesis issue., anyway, this is working fine now.

avisconti reviewed (approved):
OK, very good!!
